hydromt_fiat/fiat.py

```diff
@@ -149,100 +305,348 @@ def setup_exposure_vector(
+            list_keys.pop(0)
+
+            # add additional risk maps
+            for idx, x in enumerate(list_keys):
+                key_name = list_keys[idx]
+                layer = self.maps[key_name]
+                first_map = xr.concat([first_map, layer], dim="rp")
+
+            # convert to a dataset to be able to write attributes when writing the maps
+            # in the ouput folders. If datarray is provided attributes will not be
+            # shown in the output netcdf dataset
+            da = first_map.to_dataset(name="risk_maps")
+            da.attrs = {
+                "returnperiod": list(rp_list),
+                "type": params["map_type_lst"],
+                "name": map_name_lst,
+                "Analysis": "risk",
+            }
+            # load merged map into self.maps
+            self.set_maps(da)
+            list_maps = list(self.maps.keys())
+
+            # erase individual maps from self.maps keeping the merged map
+            for item in list_maps[:-1]:
+                self.maps.pop(item)
+
+            self.set_config("hazard.return_periods", rp_list)
+
+        # the metadata of the hazard maps is saved in the configuration toml files
+        # this component was modified to provided the element [0] od the list
+        # in case multiple maps are required then remove [0]
+        self.set_config(
+            "hazard.file",
+            [
+                str(Path("hazard") / (hazard_map + ".nc"))
+                for hazard_map in self.maps.keys()
+            ][0],
+        )
+        self.set_config(
+            "hazard.crs",
+            [
+                "EPSG:" + str((self.maps[hazard_map].rio.crs.to_epsg()))
+                for hazard_map in self.maps.keys()
+            ][0],
         )
 
-        # Store the hazard settings.
-        hazard_settings = {}
-        hazard_maps = []
-        for hazard_map in self.maps.keys():
-            hazard_maps.append(
-                str(Path("hazard") / (self.maps[hazard_map].name + ".nc"))
-            )
+        self.set_config(
+            "hazard.elevation_reference", "dem" if da_type == "water_depth" else "datum"
+        )
 
-        hazard_settings["grid_file"] = hazard_maps
+        # Set the configurations for a multiband netcdf
+        self.set_config(
+            "hazard.settings.subset",
+            [(self.maps[hazard_map].name) for hazard_map in self.maps.keys()][0],
+        )
 
-        if not isinstance(rp, list):
-            rp = "Event"
-        hazard_settings["return_period"] = rp
+        self.set_config(
+            "hazard.settings.var_as_band",
+            risk_output,
+        )
 
-        hazard_settings["crs"] = hazard.crs
-        hazard_settings["spatial_reference"] = map_type
-        self.config["hazard"] = hazard_settings
+        self.set_config(
+            "hazard.risk",
+            risk_output,
+        )
 
     def setup_social_vulnerability_index(
```

Comment:
I completely agree and also I think that it can be used already for other data, it does need to be structured in a certain way. @LiekeMeijer is this indeed the case (it can be used with other datasets)? I wrote something about it in the docstring.